File: solutions/513.find-bottom-left-tree-value/find-bottom-left-tree-value.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int findBottomLeftValue(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);
        while(!queue.isEmpty()) {
            root = queue.remove();
            if(root.right != null) 
                queue.offer(root.right);
            if(root.left != null) 
                queue.offer(root.left);
        }
        return root.val;
    }
}

File: solutions/513.find-bottom-left-tree-value/find-bottom-left-tree-value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int res = 0, h = 0;
    int findBottomLeftValue(TreeNode* root) {
        helper(root, 1);
        return res;
    }
private:
    void helper(TreeNode* root, int depth) {
        if(!root) return;
        if(h < depth) {
            res = root->val;
            h = depth;
        }
        helper(root->left, depth + 1);
        helper(root->right, depth + 1);
    }
};